linux/keyman-config/keyman_config/ibus_util.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/python3
+
+import gi
+import logging
+import subprocess
+
+gi.require_version('IBus', '1.0')
+from gi.repository import IBus, Gio
+
+def get_ibus_bus():
+    try:
+        for i in range(10000):
+            bus = IBus.Bus()
+            if bus.is_connected() and bus.is_global_engine_enabled():
+                return bus
+            bus.destroy()
+    except Exception as e:
+        logging.warning("Failed get bus")
+        logging.warning(e)
+    logging.warning("could not find connected IBus.Bus")
+    return None
+
+def install_to_ibus(bus, keyboard_id):
+    try:
+        # keyboard_id = "%s:%s" % (lang, kmx_file)
```

Comment:
I confirmed with @glasseyes the keyboard id here is namespaced **language**:**keyboard id**

This is different from the mobile apps that use **package id**:**keyboard id**

Reply:
**languagetag:full_path_to_kmx_file**
**languagetag** includes script e.g. `syr-Syrc` where the keyboard has that